packages/core/src/css/index.tsx

```diff
@@ -537,24 +537,19 @@ export function mapPropsToStyledComponent<P, T>(
     defaultVariant = 'primary',
     defaultVariants = [],
   } = options
-  const {
-    breakpoint,
-    sx,
-    theme,
-    variant = defaultVariant,
-    style,
-    variants,
-  } = props
-
-  // overrride the defaults with added ones; don't get rid of them altogether
-  let multipleVariants = [...defaultVariants]
+  const { breakpoint, sx, theme, variant, style, variants } = props
+
+  // Override the defaults with added ones; don't get rid of them altogether
+  let multipleVariants = [...defaultVariants, defaultVariant]
   if (variants?.length) {
-    multipleVariants = [...defaultVariants, ...variants]
+    multipleVariants = [...defaultVariants, defaultVariant, ...variants]
   }
+  // If a variant exists make it take precedence
+  if (variant) multipleVariants.push(variant)
```

Comment:
Maybe use the spread syntax here to stay consistent?

Reply:
Sure